app/src/main/java/com/example/dungeontest/Settings.kt

```diff
@@ -78,34 +131,83 @@ fun SettingsScreen(drawerState: DrawerState, scope: CoroutineScope) {
                     IconButton(onClick = {
                         scope.launch {
                             drawerState.open()
+                            keyboardController?.hide()
+                            focusManager.clearFocus()
                         }
                     }) {
                         Icon(Icons.Filled.Menu, contentDescription = "Open nav drawer")
                     }
                 }
             )
-        }
+        },
+        floatingActionButton = {
+            ExtendedFloatingActionButton(onClick = {
+                Log.d("SettingsScreen", "tokenValue: ${tokenValue.value.text}")
+                Log.d("SettingsScreen", "selectedCard: $selectedCard")
+                CoroutineScope(Dispatchers.IO).launch {
+                    preferences.saveToken(tokenValue.value.text)
+                    preferences.saveSelectedModel(selectedCard.value)
```

Comment:
I don't know why, but in my testing with the pixel 3a emulator, this doesn't seem to work. The selection goes back to GPT-4 after saving, even if GPT-4o was selected when I pressed save

Reply:
Also, after that, if I go to the main page and then back to the settings page, it loads with GPT-4 selected (while it does successfully load the saved value for the token field)

Reply:
Ok, this is really weird. If I try to do this without having ever rotated the screen since I started the app, the above behavior occurs. However, once I rotate the screen once (even if I then rotate it back to portrait mode), the saving of selected model works reliably, even when I exit the app and then reopen it.

Reply:
able to simulate. This happens only upon the first time you enter the app. You can try uninstalling the app and reinstalling it to clear the datastore and it will arise again. 

Im gonna need some time to dive into this.

Reply:
nvm. Think i found the root cause. 

**Running the following block of code:** 
```
CoroutineScope(Dispatchers.IO).launch { <- Coroutine 0
    preferences.saveToken(tokenValue.value.text) <- Suspending Func 1
    preferences.saveSelectedModel(selectedModel.value) <- Suspending Func 2
```
    
Will cause Suspending Func 1 and Suspending Func 2 to execute sequentially. Upon the completion of Suspending Func 1, the preferences datastore is updated and the apiToken AND _selectedModel_ variables are updated from the datastore, despite Suspending Func 2 not having completed yet. This happens because both suspending Funcs are executing in one single Coroutine
  
 **To address this synchronization issue, the code had to be modified to the following:** 
```
   CoroutineScope(Dispatchers.IO).launch { <- Coroutine 0
      launch { <- Coroutine 1
          preferences.saveToken(tokenValue.value.text) <- Suspending Func 1
      }
      launch { <- Coroutine 2
          preferences.saveSelectedModel(selectedModel.value) <- Suspending Func 2
      }

```

This ensure that both Suspending Func 1 and Suspending Func 2 can execute concurrently as they are both launched from two separate coroutines(Coroutine 1 and Coroutine 2)